components/mosart/src/riverroute/MOSART_physics_mod.F90

```diff
@@ -88,7 +94,7 @@ subroutine Euler
            if (ctlSubwWRM%ExternalDemandFlag == 0) then  ! if demand is from ELM, reset the demand0 every timestep
              StorWater%demand0(iunit) = 0
            endif
-            do nt=1,nt_rtm  
+            do nt=nt_nliq,nt_nice  
```

Comment:
The code here is error-prone as it assumes (i) there are no other tracers between liq and ice, and (ii) the index of liq tracer is before ice tracer. Please consider the following code mod:
```
do nt = 1,nt_rtm
   if (nt /= nt_nliq .and. nt /= nt_nice) cycle
   ...
enddo
```

Reply:
Good suggestion. Thanks.